### PR TITLE
feature: batch delete

### PR DESCRIPTION
**Pull request by Paul-Bob**
# Description

Fixes https://github.com/avo-hq/avo/issues/1172

batch delete

# Checklist:

- [x] I have performed a self-review of my own code
- [x] I have commented my code, particularly in hard-to-understand areas
- [ ] I have made corresponding changes to the [documentation](https://github.com/avo-hq/docs)
- [x] I have added tests that prove my fix is effective or that my feature works

### CODE REVIEW

**Comment by Paul-Bob** [comment]
@adrianthedev i don't know if any documentation is needed for this one, it's intuitive and for now we don't have any configuration to enable/disable this feature

File: lib/generators/avo/templates/locales/avo.nb-NO.yml [hunk 1]
```diff
@@ -16,6 +16,15 @@ nb-NO:
...
     resource_destroyed: 'Ressurs slettet'
+    record: # HELP NEEDED FOR TRANSLATIONS
+      zero: '%{count} records'
+      one: '%{count} record'
+      other: '%{count} records'
+    destroyed: # HELP NEEDED FOR TRANSLATIONS
+      zero: 'destroyed'
+      one: 'destroyed'
+      other: 'destroyed'
+    resource_batch_destroyed: '%{records} %{destroyed}'
```
Comment (Paul-Bob) [inline lib/generators/avo/templates/locales/avo.nb-NO.yml:27]: Can't find nobody to help us on this, want to translate it online or let it like that and wait for someone to translate it?

File: app/avo/actions/pre_made_action_batch_delete.rb [hunk 2]
```diff
@@ -0,0 +1,23 @@
...
+module Actions
+  class PreMadeActionBatchDelete < Avo::BaseAction
+    self.name = I18n.t('avo.delete').capitalize
+    self.message = I18n.t('avo.are_you_sure').capitalize
+    self.confirm_button_label = I18n.t('avo.delete').capitalize
+
+    def handle(**args)
+      models, = args.values_at(:models, :fields, :current_user, :resource)
+
+      models.each(&:destroy!)
```
Comment (veelenga) [inline app/avo/actions/pre_made_action_batch_delete.rb:12]: This should probably be wrapped into transaction

Comment (Paul-Bob) [reply app/avo/actions/pre_made_action_batch_delete.rb:12]: Actually i thanked about 3 approaches and i'm not sure which one is the better one.

1 - As you suggested wrap the delete into transaction and if one fails none is deleted
2 - Output the exception to the user for each model that fails 
3 - Delete all and stop when one of them fails

I think the better one is the one you suggested, @adrianthedev what do you think?

File: lib/generators/avo/templates/locales/avo.nb-NO.yml (hunk 1 above)
Comment (adrianthedev) [reply lib/generators/avo/templates/locales/avo.nb-NO.yml:27]: Try and find a similar project that has translations (think Wordpress, e-commerce websites, etc.) or use Google translate. We'll get suggestions from developers if they are wrong.

File: app/avo/actions/pre_made_action_batch_delete.rb (hunk 2 above)
Comment (adrianthedev) [reply app/avo/actions/pre_made_action_batch_delete.rb:12]: I'm now thinking about exposing this action somehow to the public as it is and allowing the developer to choose the strategy.

I mean, when you run `avo:install` to generate this action in the Rails app and then enable the developer to add it to their resources and modify it as they wish.

The downside is that we should add the `index_controls` option too.

**Comment by Paul-Bob** [comment]
The batch destroy logic approach should be decided by the developer for each project, closing this for now.

**Pull request closed without merging**